File: app/src/main/java/com/openclassrooms/realestatemanager/models/AddressProperties.java
package com.openclassrooms.realestatemanager.models;

import androidx.room.Entity;
import androidx.room.ForeignKey;

@Entity(
        primaryKeys = {"mPropertyId", "mAddressId"},
        foreignKeys = {
                @ForeignKey( entity = Property.class,
                        parentColumns = "mId",
                        childColumns = "mPropertyId"),
                @ForeignKey(entity = Address.class,
                        parentColumns = "mId",
                        childColumns = "mAddressId")
        }
)
public class AddressProperties {

    private long mPropertyId; // PK|FK
    private long mAddressId; // PK|FK

    public AddressProperties(long propertyId, long addressId) {
        mPropertyId = propertyId;
        mAddressId = addressId;
    }

    public long getPropertyId() {
        return mPropertyId;
    }

    public void setPropertyId(long propertyId) {
        mPropertyId = propertyId;
    }

    public long getAddressId() {
        return mAddressId;
    }

    public void setAddressId(long addressId) {
        mAddressId = addressId;
    }
}

File: app/src/main/java/com/openclassrooms/realestatemanager/models/AddressProperties.kt
package com.openclassrooms.realestatemanager.models

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index

@Entity(indices = [Index("propertyId"), Index("addressId")],
        primaryKeys = ["propertyId", "addressId"],
        foreignKeys =
            [ForeignKey(entity = Property::class, parentColumns = ["id"], childColumns = ["propertyId"]),
            ForeignKey(entity = Address::class, parentColumns = ["id"], childColumns = ["addressId"])])
class AddressProperties(var propertyId: Long // PK|FK
                        , var addressId: Long // PK|FK
)
